api/src/main/java/org/openmrs/module/ohricore/api/impl/HIVStatusComputedConcept.java

```diff
@@ -0,0 +1,72 @@
+package org.openmrs.module.ohricore.api.impl;
+
+import org.openmrs.Concept;
+import org.openmrs.Encounter;
+import org.openmrs.EncounterType;
+import org.openmrs.Obs;
+import org.openmrs.Patient;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.ohricore.api.OHRIComputedConcept;
+import org.openmrs.module.ohricore.engine.CommonsUUID;
+import org.openmrs.module.ohricore.engine.HIVStatusConceptUUID;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+/**
+ * @author MayanjaXL, Amos, Stephen, smallGod date: 28/06/2021
+ */
+@Component("hivStatusComputedConcept")
+public class HIVStatusComputedConcept implements OHRIComputedConcept {
+	
+	@Override
+	public Obs compute(Encounter triggeringEncounter) {
+		
+		Concept hivFinalTestConcept = getConcept(HIVStatusConceptUUID.FINAL_HIV_TEST_RESULT);
+		
+		Patient patient = triggeringEncounter.getPatient();
+		List<Obs> hivTestObs = Context.getObsService().getObservationsByPersonAndConcept(patient.getPerson(),
+		    hivFinalTestConcept);
+		
+		Concept hivStatus = computeHivStatusConcept(hivTestObs);
+		return createOrUpdateObs(patient, hivStatus);
+	}
+	
+	private Concept computeHivStatusConcept(List<Obs> hivTestObs) {
+
+        Supplier<Stream<Obs>> hivTestObsStream  = hivTestObs::stream;
+        return hivTestObsStream.get()
+                .filter(obs -> obs.getValueCoded() == getConcept(CommonsUUID.POSITIVE))
+                .findAny()
+                .map(Obs::getValueCoded)
+                .orElse(hivTestObsStream.get()
+                        .filter(obs -> obs.getValueCoded() == getConcept(CommonsUUID.NEGATIVE))
+                        .filter(obs -> valueDateIsWithin90Days(obs.getValueDate()))
+                        .findAny()
+                        .map(Obs::getValueCoded)
+                        .orElse(getConcept(CommonsUUID.UNKNOWN))
+                );
+    }
+	
+	@Override
+	public EncounterType getTargetEncounterType() {
```

Comment:
Why is the target encounter type null?

Reply:
We are not using a target Encounter for now as earlier thought. So some of its methods are still, we might have to remove it/them or not. Kind'a early to tell now.

Reply:
@smallgod Do not add functions that you do not need now

Reply:
Okay, deleted